config/initializers/form_errors.rb

```diff
@@ -0,0 +1,15 @@
+# Add error span to form fields with errors
+ActionView::Base.field_error_proc = proc do |html_tag, instance|
+  if /<label/.match?(html_tag)
+    # Remove text color if it exists so that container color takes precedence
```

Comment:
I played around with a couple options. Originally, I wanted to avoid regex and just use a container. I tried putting `text-red-600!` on the container, but of course the specificity of having text color classes on the label directly outweighed this.

I also debated swapping any text coloring on the label element class, but figured there'd have to be way more logic to handle cases where there aren't any class attributes. Ultimately, clearing any text colors then adding a container seemed to be the shortest and most inclusive solution. 

Let me know if there is a better way!

Reply:
I like this approach, I just wish it wasn't so much code. It took me a few minutes to wrap my head around what is going on.

Can you please...

1. Add a comment on line 8 explaining what is going on? Similar to the comment on line 4.
2. Extract the actual algorithm to a Ruby model, then call said model from this initializer.
2. Add an integration test that verifies `text-red-600` is added to a label when trying to submit an invalid form.
3. Add a unit test for the new PORO that passes in an HTML block and verifies the result is correct.

Reply:
I may have gone overboard with the comments, I just wanted to be sure to save time for any one trying to read this code. I'm happy to remove any unnecessary comments